README.md

```diff
@@ -12,23 +12,359 @@
+In the near future, testlab will include its own terraform recipes in the
+`automation` directory.
+
+## Installation
+
+Testlab is a simple go binary, and can be installed into your GOPATH as such:
+
+```
+$ go get github.com/libp2p/testlab/testlab
+```
+
+## How it Works
+
+Testlab is an automation layer over Hashicorp's
+[Nomad](https://nomadproject.io), a cluster manager in the same style as
+Kubernetes. Testlab's primary goal is to make it simple to launch large clusters
+of peer-to-peer applications to better understand how they function at scale.
+
+Testlab topologies are built around two main concepts: peer deployments and
+scenario runners. Generally, a peer deployment describes a set of instances of a
+peer-to-peer application and, optionally, how they are connected. A scenario
+runner is a special program launched in the cluster that can remotely control
+peer deployments to simulate activity within the network.
+
+The goal output of a testlab topology is metrics data. While, in the future,
+it would be nice to support correctness tests, the current aim is to allow for
+large scale benchmarking and diagnosis of issues, as well as regression
+testing. All peer deployments should be instrumented with prometheus-friendly
+metrics, should they want to have data collected. This is described in greater
+detail in the [scenario runners](#scenario-runners) section. Clusters
+specifying a deployment of the `prometheus` plugin will automatically have
+metrics collected.
+
+## Usage
+
+Testlab is a wrapper over [nomad](https://www.nomadproject.io/)'s golang API,
+making it easy to deploy pre-configured networks of p2p applications.
+
+Most users of testlab need only concern themselves with two concepts, the
+[deployment configuration](#deployment-configuration), and
+[scenarios](#scenario-runners). Users wishing to add testlab support for their
+own daemons will need to understand the [node API](#node-api), as well.
+
+### CLI
+
+The testlab CLI depends on the presence of the standard environment variables to
+connect out to your Nomad and Consul clusters. If any are ommitted, the
+defaults, as defined by Hashicorp, will be applied. The defaults are typically
+usable in development.
+
+- [Nomad environment variables](https://github.com/hashicorp/nomad/blob/v0.9.3/api/api.go#L217)
+- [Consul environment variables](https://github.com/hashicorp/consul/blob/v1.5.2/api/api.go#L24)
+
+Furthermore, users can optionally provide a path in the environment variable
+`TESTLAB_ROOT` to define where the testlab metadata will be stored. This
+defaults to `/tmp/testlab`. **NOTE**: In order to have multiple testlab
+topologies in flight at the same time, one must define different `TESTLAB_ROOT`s
+for each topology.
+
+The testlab CLI has two commands:
+
+- `testlab start <json configuration>`
+  Parses, evaluates for correctness, and attempts to deploy a topology as
+  defined by the provided json configuration file.
```

Comment:
And then attempt to run the `scenarios`?

Reply:
technically the scenarios are part of the topology, but i'll mention that.